commit c36ed165b5c96da72af34373eb9f56c550f08d96
Author: Dag Haavi Finstad <daghf@varnish-software.com>
Date:   Fri Nov 24 16:13:08 2017 +0100

    Drop VBT_Wait call in vbe_dir_finish
    
    With VBT_Close now being capable of dealing with STOLEN connections, we
    no longer need to VBT_Wait for them prior to close.

Index: varnish4/bin/varnishd/cache/cache_backend.c
===================================================================
--- varnish4.orig/bin/varnishd/cache/cache_backend.c
+++ varnish4/bin/varnishd/cache/cache_backend.c
@@ -158,13 +158,15 @@ vbe_dir_finish(const struct director *d,
 	CAST_OBJ_NOTNULL(vbc, bo->htc->priv, VBC_MAGIC);
 	bo->htc->priv = NULL;
 	if (vbc->state != VBC_STATE_USED)
-		VBT_Wait(wrk, vbc);
+		assert(bo->htc->doclose == SC_TX_PIPE);
 	if (bo->htc->doclose != SC_NULL || bp->proxy_header != 0) {
 		VSLb(bo->vsl, SLT_BackendClose, "%d %s", vbc->fd,
 		    bp->display_name);
 		VBT_Close(bp->tcp_pool, &vbc);
+        AZ(vbc);
 		Lck_Lock(&bp->mtx);
 	} else {
+        assert (vbc->state == VBC_STATE_USED);
 		VSLb(bo->vsl, SLT_BackendReuse, "%d %s", vbc->fd,
 		    bp->display_name);
 		Lck_Lock(&bp->mtx);
